actions/views/channel.test.js

```diff
@@ -103,6 +104,9 @@ describe('channel view actions', () => {
                 loadingPosts: {},
                 postVisibility: {current_channel_id: 60},
             },
+            rhs: {
+                selectedPostId: '',
```

Comment:
Could you add a couple tests for this new behaviour to ensure it doesn't get broken accidentally in the future? They could be unit tests in here or E2E tests

Reply:
Now that we are at it, I want to bring something on the table, 
we have many snapshot tests, that are not testing anything at all...
Snapshot tests IMO should use the `react-test-renderer` and never `shallow` mount.

e.g: https://github.com/mattermost/mattermost-webapp/blob/a468fdb80e61c876462ed56f133c562fc95db737/components/rhs_thread/__snapshots__/rhs_thread.test.tsx.snap#L3-L40

The above test would have caught the regression, but did not.
I get that mounting components requires extra work, but I'm positive that it would be beneficial.

Reply:
I've added a new e2e test for leaving a channel while RHS is open.
@jgilliam17 can you create a zephyr test case for 
https://github.com/mattermost/mattermost-webapp/pull/9495/files#diff-c308c925a9f30488189bdc1fa95155f822d2135aad3abe2d34ffa95b1fb07640R38

Reply:
I don't know too much about `react-test-renderer`, but I assume it's similar to Enzyme's `mount`. Is that correct?

Regardless, I'm not a fan of snapshot tests myself since I find they either break due to unrelated changes (especially when using snapshots and `mount` together) or they make don't make it clear what they're looking for. I do like using `mount` and the enzyme functionality as a lighter weight version of what the E2E tests provide though.

Reply:
`react-test-renderer` is like `mount` but also offers a `toJSON`[2] so that snapshot[2] testing makes sense.
Searching a bit it seems there is a plugin[3] for `enzyme` to offer serialize a component to JSON, making snapshot output a little more meaningful. 

[1]: jest snapshot testing https://jestjs.io/docs/snapshot-testing#snapshot-testing-with-jest
[2]: react-test-renderer https://reactjs.org/docs/test-renderer.html#testrenderertojson
[3]: enzyme-to-json https://github.com/adriantoine/enzyme-to-json